src/main/java/lotto/domain/Lotto.java

```diff
@@ -1,41 +1,46 @@
 package lotto.domain;
 
-import lotto.util.LottoNumberGenerator;
+import lotto.constants.Rank;
+import lotto.util.InputValidator;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class Lotto {
 
     private static final int COLLECT_ADD_NUMBER = 1;
     private static final int NOT_COLLECT_ADD_NUMBER = 0;
 
-    private final List<Integer> lottoNumbers;
+    private final List<LottoNumber> lottoNumbers;
 
-    public Lotto() {
-        lottoNumbers = LottoNumberGenerator.generateLottoNumbers();
+    public Lotto(List<LottoNumber> lottoNumbers){
+        this.lottoNumbers = lottoNumbers;
     }
 
-    public Lotto(List<Integer> lottoNumbers) {
-        this.lottoNumbers = lottoNumbers;
+    public static Lotto of(List<Integer> lottoNumbers){
+        InputValidator.validateLottoNumberCount(lottoNumbers.size());
+        InputValidator.validateDuplicateLottoNumber(lottoNumbers);
+        return new Lotto(lottoNumbers.stream().map(LottoNumber::new).collect(Collectors.toList()));
```

Comment:
```suggestion
        return new Lotto(lottoNumbers.stream()
          .map(LottoNumber::new)
          .collect(Collectors.toList()));
```

적절한 개행처리 또한 코딩컨벤션일수 있습니다.
- 가독성향상
- 디버깅시 이점

Reply:
넵! 적절히 개행하도록 하겠습니다.